### DISCUSSION TITLE
Web3modal + custom JSON RPC request (EIP-1193)

### DISCUSSION THREAD

**Discussion by Sednaoui** [wevm/wagmi#2706] [Q&A]
Hi! 

I am looking to send a custom rpc request to a wallet connected through web3modal / walletconnect. 
```js
const customRequest = {
    method: "wallet_customRequest",
    params: [..]
}
```
I am trying through  the `useWalletClient()` hook like the following: 
```js
import { useWalletClient } from 'wagmi"

const { data: walletClient } = useWalletClient();
..
await walletClient.request(customRequest);
```

Now the problem is, useWalletClient enforces the request to be called to the default rpc provider. This method is a wallet specific method and not a public method. So the request is being rejected with an error. I am not sure why the request is being called to the public rpc provider and not sent directly to the wallet. 

I went and tried directly with the viem method `createWalletClient`. I was able to find examples on how to set the transport http to an injected wallet. 
```js
const client = createWalletClient({
  chain: mainnet,
  transport: custom(window.ethereum)
})
```
Now is there a way to set the transport to the web3modal provider so I am able to send this request directly to the wallet? 

Thank you

**Comment by Sednaoui** [comment]
To further add, sending requests with methods like `wallet_switchEthereumChain` works fine with useWalletClient Hook.

```js
const { data: walletClient } = useWalletClient();

await walletClient.request({
  method: "wallet_switchEthereumChain",
  params: [{ chainId: 0x5 }],
});
```

But custom methods have problems going through, even if the wallet supports this specific method.

**Comment by tcheee** [comment]
Hey, I have the same issue as I want to integrate with 4337 wallet and enable a bundle call, did you find any solution?
